File: AndroidAppBaseLibrary/src/main/java/stark/android/appbase/base/BaseActivity.java
package stark.android.appbase.base;

import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

/**
 * Created by jihongwen on 16/9/8.
 */

public abstract class BaseActivity extends AppCompatActivity {


    public void startFragment(Fragment fragment, int resId, String tag) {
        getSupportFragmentManager().beginTransaction()
                .add(resId, fragment, tag).commit();
    }

    public Fragment findFragment(String tag) {
        return getSupportFragmentManager().findFragmentByTag(tag);
    }
}

File: AndroidAppBaseLibrary/src/main/java/stark/android/appbase/activity/BaseActivity.kt
package stark.android.appbase.activity

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import stark.android.appbase.R

/**
 * Created by jihongwen on 16/9/8.
 */

open class BaseActivity : AppCompatActivity() {

    fun startFragment(fragment: Fragment, tag: String) {
        startFragment(fragment, R.id.fragment_container, tag)
    }

    fun startFragment(fragment: Fragment, resId: Int, tag: String) {
        supportFragmentManager.beginTransaction()
                .replace(resId, fragment, tag).commit()
    }

    fun findFragment(tag: String): Fragment? {
        return supportFragmentManager.findFragmentByTag(tag)
    }
}
